### PR TITLE
add embedded zk test

### PR DESCRIPTION
**Pull request by yulunli**

### CODE REVIEW

File: zk/client_test.go [hunk 1]
```diff
@@ -56,6 +56,15 @@ func (s *ZKClientTestSuite) TearDownTest() {
 	}
 }
 
+func (s *ZKClientTestSuite) TestEmbeddedZk() {
+	err := EnsureZookeeperUp(s.EmbeddedZkPath)
+	s.NoError(err)
+	err = StopZookeeper(s.EmbeddedZkPath)
+	s.NoError(err)
+	err = EnsureZookeeperUp(s.EmbeddedZkPath)
```
Comment (zhijinli) [inline zk/client_test.go]: is this to test reboot? should we stop in the end?

File: Makefile [hunk 2]
```diff
@@ -35,7 +35,7 @@ endif
 
 .PHONY: test
 test:
-	go test -race $(PKGS)
+	go test -p 1 -race $(PKGS)
```
Comment (zhijinli) [inline Makefile]: what does this do and why is this needed?

Comment (yulunli) [reply Makefile]: We don't want multiple suites to run in parallel since we have a test to restart zookeeper.

File: zk/client_test.go (hunk 1 above)
Comment (yulunli) [reply zk/client_test.go]: Just want to make sure the cluster can start after stop

File: Makefile (hunk 2 above)
Comment (zhijinli) [reply Makefile]: is it possible to blacklist parallel run for that particular test instead?

File: zk/client_test.go (hunk 1 above)
Comment (zhijinli) [reply zk/client_test.go]: I was wondering is it a good practice to always stop after start. something to think about, non-blocking.

File: Makefile (hunk 2 above)
Comment (yulunli) [reply Makefile]: No tests in a suite are parallel by default. https://github.com/stretchr/testify/issues/52
